src/test/java/org/elasticsearch/percolator/PercolatorTests.java

```diff
@@ -2034,5 +2029,25 @@ public void testNestedDocFilter() throws IOException {
                 .get();
         assertMatchCount(response, 3l);
     }
+
+    @Test
+    public void testMapUnmappedFieldAsString() throws IOException{
+        // If index.percolator.map_unmapped_fields_as_string is set to true, unmapped field is mapped as an analyzed string.
+        ImmutableSettings.Builder settings = ImmutableSettings.settingsBuilder()
+                .put(indexSettings())
+                .put("index.percolator.map_unmapped_fields_as_string", true);
+        assertAcked(prepareCreate("test")
+                .setSettings(settings));
+        client().prepareIndex("test", PercolatorService.TYPE_NAME)
+                .setSource(jsonBuilder().startObject().field("query", queryStringQuery("VALUE")).endObject()).get();
```

Comment:
maybe just use a match query on field `field1` with value `value`, to make it clearer that this is query on field, that would normally fail?